spec/api/user_spec.rb

```diff
@@ -428,11 +427,13 @@ def add_flags(content, expected_result)
             end
           end
         end
+        # Sort the map entries using the default sort
+        expected_result = expected_result.values.sort_by  { |val| [val["threads"], val["responses"], val["replies"]] }.reverse
 
-        get "/api/v1/users/#{course_id}/stats"
+        get "/api/v1/users/#{course_id}/stats2"
         expect(last_response.status).to eq(200)
         res = parse(last_response.body)
-        expect(res).to eq expected_result
+        expect(res["user_stats"]).to eq expected_result
```

Comment:
The new API now returns a list of objects sorted by activity but other than that the results should be the same as before. 

I'll add more tests for checking that adding comments, removing comments, etc keep the counts updated.